### PR TITLE
OpenID4VCI: Retrieve credentials synchronously

### PR DESCRIPTION
**Pull request by reinkrul**
To put responsibility of retrying on offerer.

Fixes https://github.com/nuts-foundation/nuts-node/issues/2040

### CODE REVIEW

File: vcr/holder/oidc4vci_wallet.go [hunk 1]
```diff
@@ -113,53 +113,60 @@ func (h wallet) HandleCredentialOffer(ctx context.Context, offer oidc4vci.Creden
...
-	// TODO: store offer and perform these requests async
-	//       See https://github.com/nuts-foundation/nuts-node/issues/2040
 	accessTokenResponse, err := issuerClient.RequestAccessToken(oidc4vci.PreAuthorizedCodeGrant, map[string]string{
 		"pre-authorized_code": preAuthorizedCode,
 	})
 	if err != nil {
-		return fmt.Errorf("unable to request access token: %w", err)
+		return oidc4vci.Error{
+			Err:        fmt.Errorf("unable to request access token: %w", err),
+			Code:       oidc4vci.InvalidToken,
```
Comment (gerardsn) [inline vcr/holder/oidc4vci_wallet.go:129]: it is weird to receive a InvalidToken error on credential offer, but probably fine in a temporary situation?

**Review by gerardsn — COMMENTED** [review]
The issue mentions:
> - We'll add the custom response to the Nuts OpenID4VCI RFC

Is the custom response documented anywhere?

**Review by woutslakhorst — APPROVED** [review]

**Comment by woutslakhorst** [comment]
> The issue mentions:
> 
> > * We'll add the custom response to the Nuts OpenID4VCI RFC
> 
> Is the custom response documented anywhere?

should be added to https://github.com/nuts-foundation/nuts-specification/pull/257

**Comment by reinkrul** [comment]
I added a TODO in that PR to add it.

**Pull request merged**